plugins/pulp_python/plugins/importers/importer.py

```diff
@@ -181,11 +183,10 @@ def upload_unit(self, repo, type_id, unit_key, metadata, file_path, conduit, con
         :rtype:           dict
         """
         package = models.Package.from_archive(file_path)
-        package.init_unit(conduit)
-
-        shutil.move(file_path, package.storage_path)
-
-        package.save_unit(conduit)
+        package.save()
+        package.import_content(file_path)
+        repo_controller.associate_single_unit(repo.repo_obj, package)
+        repo_controller.rebuild_content_unit_counts(repo.repo_obj)
```

Comment:
Good idea. Since @pcreech is working on doing this in the platform now, I'm going to let him handle it there.